### DISCUSSION TITLE
v5.1.2  域名解析返回 REFUSED

### DISCUSSION THREAD

**Discussion by PHCSJC** [IrineSistiana/mosdns#634] [Q&A]
退回到v5.1.1可以了

**Comment by Mrican** [comment]
v5.1.2可以啊
![微信图片_20230305083733](https://user-images.githubusercontent.com/73725611/222935341-74e357f2-cf1b-4106-affd-5ca715301d96.png)

**Comment by PHCSJC** [comment]
今天正好有时间，就又复测了一遍，上面的问题依然存在，可能是在我的配置下 #605 才会出现这个bug，但我的配置本身肯定没问题，已经稳定运行1个多月了，而且这个问题我只要换回上个5.1.1版本就正常了

平台：amd64 linux和arm64都可复现
问题：解析为空：
dig gost.run

; <<>> DiG 9.11.5-P4-5.1+deb10u7-Debian <<>> gost.run
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: REFUSED, id: 15519
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;gost.run.			IN	A

;; Query time: 7 msec
;; SERVER: 192.168.2.1#53(192.168.2.1)
;; WHEN: Mon Mar 06 08:32:09 CST 2023
;; MSG SIZE  rcvd: 26

**Reply by urlesistiana** [reply]
#621 

你的请求是不是有 ecs？

**Reply by PHCSJC** [reply]
> #621
> 
> 你的请求是不是有 ecs？

没有哈，dig不带任何参数，或者win上的nslookup

**Comment by PHCSJC** [comment]
v5.1.3还是有这个问题，域名解析失败，退回v5.1.1瞬间恢复

**Comment by Mrican** [comment]
; <<>> DiG 9.16.30 <<>> gost.run
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 64507
;; flags: qr rd ra; QUERY: 1, ANSWER: 2, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 1232
; COOKIE: e3a1dacf0d91c9cd0100000064127952187496233b47cb80 (good)
;; QUESTION SECTION:
;gost.run.                      IN      A

;; ANSWER SECTION:
gost.run.               300     IN      A       104.21.58.195
gost.run.               300     IN      A       172.67.163.116

;; Query time: 350 msec
;; SERVER: 127.0.0.1#53(127.0.0.1)
;; WHEN: Thu Mar 16 10:05:04 中国标准时间 2023
;; MSG SIZE  rcvd: 113
![微信图片_20230316101206](https://user-images.githubusercontent.com/73725611/225491909-31bd0919-5ee8-4a91-a70c-a7f90e1c7ab3.png)

**Comment by urlesistiana** [comment]
~~你的某个上游返回了 REFUSE 。这个 REFUSE  应该是这个域名的源服务器返回的。源服务器返回这个信号有它的目的。~~

v5.1.2 以后的版本会遵守 DNS 转发器的规范，默认不过滤这些信号。

以后的版本可能会支持设定过滤方式。

**Reply by PHCSJC** [reply]
上游是8.8.8.8和1.1.1.1，不至于2个都refuse吧
我试了，不至这个域名，只要是不在列表里的域名，都是无法解析，所以应该还是fallback这块的逻辑有哪里过不去

**Reply by urlesistiana** [reply]
确实是你的配置问题。local_sequence 那个插件不要用 reject，用 drop_resp 。

**Reply by PHCSJC** [reply]
原来是这里的问题，可以了，非常感谢！
